ui/components/multichain/asset-picker-amount/asset-picker-modal/asset-picker-modal.tsx

```diff
@@ -111,6 +123,26 @@ export function AssetPickerModal({
 
   const isDest = sendingAssetImage && sendingAssetSymbol;
 
+  const chainId = useSelector(getCurrentChainId);
+
+  const nativeCurrencyImage = useSelector(getNativeCurrencyImage);
+  const nativeCurrency = useSelector(getNativeCurrency);
+  const balanceValue = useSelector(getSelectedAccountCachedBalance);
+
+  const tokenConversionRates = useSelector(getTokenExchangeRates, isEqual);
+  const conversionRate = useSelector(getConversionRate);
+  const currentCurrency = useSelector(getCurrentCurrency);
+
+  const {
+    currency: primaryCurrency,
+    numberOfDecimals: primaryNumberOfDecimals,
+  } = useUserPreferencedCurrency(PRIMARY, { ethNumberOfDecimals: 4 });
+
+  const [, primaryCurrencyProperties] = useCurrencyDisplay(balanceValue, {
+    numberOfDecimals: primaryNumberOfDecimals,
+    currency: primaryCurrency,
+  });
```

Comment:
Is there another util we can use to get this value? When the user's balance is small, `primaryCurrencyProperties.value` is a human-readable string like `< 0.000001`, which causes `getTokenFiatAmount` to throw an error

Reply:
Uploading Screen Recording 2024-05-10 at 2.36.12 PM.mov…

Reply:
https://github.com/MetaMask/metamask-extension/assets/100321200/a45c4f1f-7c0b-4260-89ce-3ef6249fc956

Reply:
This error was related to an issue in a pair of utils we depend on; updated in 0965dd66a42a0ebef8829f7d0fe2e82f8a146408